src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1610,12 +1617,13 @@ private async ValueTask<Stream> EstablishSocksTunnel(HttpRequestMessage request,
             return stream;
         }
 
-        private void HandleHttp11ConnectionFailure(HttpRequestMessage request, Exception e)
+        private void HandleHttp11ConnectionFailure(TaskCompletionSourceWithCancellation<HttpConnection>? requestWaiter, Exception e)
         {
             if (NetEventSource.Log.IsEnabled()) Trace($"HTTP/1.1 connection failed: {e}");
 
-            bool failRequest;
-            TaskCompletionSourceWithCancellation<HttpConnection>? waiter;
+            // If this is happening as part of an HTTP/2 => HTTP/1.1 downgrade, we won't have an HTTP/1.1 waiter associated with this request
+            requestWaiter?.TrySetException(e);
```

Comment:
I think it'd be worth a comment explaining why we don't care about the return value, which presumably is because the associated request may have already been processed by another connection?